### PR TITLE
getTag method returns MapData

### PR DESCRIPTION
**Pull request by friendlyhj**

### CODE REVIEW

File: src/main/java/com/blamejared/crafttweaker/api/item/IItemStack.java [hunk 1]
```diff
@@ -437,12 +437,12 @@ default boolean hasTag() {
...
      *
-     * @return IData of the ItemStack NBT Tag, null if it doesn't exist.
+     * @return MapData of the ItemStack NBT Tag, null if it doesn't exist.
      */
     @ZenCodeType.Getter("tag")
-    default IData getTag() {
+    default MapData getTag() {
         
-        return NBTConverter.convert(getInternal().getTag());
+        return new MapData(getInternal().getTag());
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/crafttweaker/api/item/IItemStack.java]: This doesn't take into account if the tag is null, which NBT converter did.

**Review by jaredlll08 — CHANGES_REQUESTED** [review]
This is also a breaking change, if a mod was calling this method before, they would need to recompile against the new code since the method signature would have changed.

File: src/main/java/com/blamejared/crafttweaker/api/item/IItemStack.java [hunk 2]
```diff
@@ -373,270 +359,283 @@ default boolean isDamaged() {
...
+     * @return MapData of the ItemStack NBT Tag, null if it doesn't exist.
      */
     @ZenCodeType.Getter("tag")
-    default IData getTag() {
-        
+    @ZenCodeType.Nullable
+    default MapData getTagNew() {
+
+        CompoundNBT tag = getInternal().getTag();
+        return tag == null ? null : new MapData(tag);
```
Comment (kindlich) [inline src/main/java/com/blamejared/crafttweaker/api/item/IItemStack.java]: Why not just create an overload in NBTConverter like
```java
public static MapData convert(CompoundNBT compound) {
    return compound == null ? null : new MapData(compound);
}

public static IData convert(INBT nbt) {
    if(nbt == null) {
        return null;
    }

    switch(nbt.getId()) {
       //...
        case 10:
            return convert((CompoundNBT)nbt);
        //...
    }
}

File: src/main/java/com/blamejared/crafttweaker/api/item/IItemStack.java [hunk 3]
```diff
@@ -373,270 +359,283 @@ default boolean isDamaged() {
...
-    default IData getOrCreateTag() {
-        
-        if(getInternal().getTag() == null) {
+    default MapData getOrCreateTagNew() {
+
+        if (getInternal().getTag() == null) {
             getInternal().setTag(new CompoundNBT());
         }
-        return NBTConverter.convert(getInternal().getTag());
+        return new MapData(getInternal().getTag());
```
Comment (kindlich) [inline src/main/java/com/blamejared/crafttweaker/api/item/IItemStack.java]: If the suggestion of an overload is accepted, then you can just keep NBTConverter.convert here ^^

**Comment by friendlyhj** [comment]
Close it. Since we can auto down cast now. The last thing the PR would add is to mark `@ZenCodeType.Nullable` on `MapData#getAt` and `MapData#put` method. It is too easy. I think you can directly make a commit about it.

**Pull request closed without merging**